camayoc/qcs_models.py

```diff
@@ -287,3 +291,127 @@ def equivalent(self, other):
+
+    Example::
+        >>> from camayoc.qcs_models import NetworkProfile, HostCredential, Scan
+        >>>
+        >>> hostcred = HostCredential(password='foo')
+        >>> hostcred.create()
+        >>> netprof = NetworkProfile(hosts=['0.0.0.0'],
+                                     credential_ids=[hostcred._id])
+        >>> netprof.create()
+        >>> scan = Scan(profile_id=netprof._id)
+        >>> scan.create()
+        >>> scan.pause()
+        >>> assert scan.status() == 'paused'
+    """
+
+    def __init__(
+            self,
+            client=None,
+            profile_id=None,
+            max_concurrency=50,
+            scan_type='host',
+            _id=None):
+        """Iniitalize a Scan object with given data.
+
+        If no value for max_concurrency is given, it will be set to 50, and if
+        no value for scan_type is given, the default type is host (for which
+        facts are collected). The other valid option for scan_type is
+        'discovery' in which no facts are collected, the server simply sees how
+        many hosts in the profile it can make contact and log into.
+        """
+        super().__init__(client=client, name=None, _id=_id)
+
+        # scan jobs do not have a name, so we delete this attribute
+        del self.name
+        self.profile = profile_id
+        self.endpoint = QCS_SCAN_PATH
+
+        # valid scan types are 'host' and 'discovery'
+        self.scan_type = scan_type
+        self.max_concurrency = max_concurrency
+
+    def delete(self):
+        """No delete method is implemented for scan objects.
+
+        Emit a warning to the camayoc user instead of doing anything.
+        """
+        LOGGER.critical(
+            'Camayoc Warning: The DELETE method is not allowed for scans')
+
+    def pause(self, **kwargs):
+        """Send PUT request to self.endpoint/{id}/pause/ to pause a scan.
+
+        :param ``**kwargs``: Additional arguments accepted by Requests's
+            `request.request()` method.
+        """
+        path = urljoin(self.path(), 'pause/')
+        return self.client.put(path, {}, **kwargs)
+
+    def cancel(self, **kwargs):
+        """Send PUT request to self.endpoint/{id}/cancel/ to cancel a scan.
+
+        :param ``**kwargs``: Additional arguments accepted by Requests's
+            `request.request()` method.
+        """
+        path = urljoin(self.path(), 'cancel/')
+        return self.client.put(path, {}, **kwargs)
+
+    def restart(self, **kwargs):
+        """Send PUT request to self.endpoint/{id}/restart/ to restart a scan.
+
+        :param ``**kwargs``: Additional arguments accepted by Requests's
+            `request.request()` method.
+        """
+        path = urljoin(self.path(), 'restart/')
+        return self.client.put(path, {}, **kwargs)
+
+    def status(self):
+        """Check on the status of the scan.
+
+        Note: only scans that have been posted to the server have a status.
+        """
+        if not self._id:
```

Comment:
So the ID gets set if the user calls create() and it is successful...basically the self.read() is going to fail if the scan has yet to be created or if the  create() failed.

My thought was that a scan that doesn't exist on the server doesn't really have a status so that is why I was returning None. I could instead just let the HTTPError that you get from trying to read something that is not on the server bubble up to the user.